Packages/cdms2/Lib/cdmsobj.py

```diff
@@ -549,12 +546,12 @@ def matchone(self, pattern, attname):
         """
         if attname is None:
             for attval in self.attributes.values():
-                if type(attval) is types.StringType and pattern.match(attval) is not None:
+                if isinstance(attval,basestring) and pattern.search(attval) is not None:
```

Comment:
This has a different behavior: match vs search. Not sure what the correct behavior is.